pulsar-client/src/main/java/org/apache/pulsar/client/api/ActiveConsumerListener.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.client.api;
+
+/**
+ * Listener on the consumer state changes.
+ */
+public interface ActiveConsumerListener {
```

Comment:
If we use a single method, it becomes accessible through a lambda. Eg: 

```java
interface ActiveConsumerListener {
   void statusChanged(Consumer, int partitionId, Status status);
}
```

Reply:
I think lambda is good for non-branchable operations. However it doesn't fit well for branchable operations. we can make a single method here using `Status`, however the application logic has to write `if-else` block to take different actions. It is much clear to me if we have two different methods for two different actions. so I chose to have separated methods for handling become active and become inactive, when applications are wring the listener, you just need to think what should do when it become active and what should do when it become inactive.

However if you think using `Status` and having a lambda interface is much better, let me know. I will make the change.

Reply:
Sure, that makes sense. 

One other thing that I was thinking is that we could have this listener to be more general oriented, so that we could in future reuse it for other kinds of notifications. 

Something like `ConsumerEventsListener` so that is more neutral? What do you think?